experimental/LieAlgebras/src/AbstractLieAlgebra.jl

```diff
@@ -0,0 +1,155 @@
+@attributes mutable struct AbstractLieAlgebra{C<:RingElement} <: LieAlgebra{C}
+  R::Ring
+  dim::Int
+  struct_consts::Matrix{SRow{C}}
+  s::Vector{Symbol}
+
+  function AbstractLieAlgebra{C}(
+    R::Ring,
+    struct_consts::Matrix{SRow{C}},
+    s::Vector{Symbol};
+    cached::Bool=true,
+    check::Bool=true,
+  ) where {C<:RingElement}
+    return get_cached!(
+      AbstractLieAlgebraDict, (R, struct_consts, s), cached
+    ) do
+      (n1, n2) = size(struct_consts)
+      @req n1 == n2 "Invalid structure constants dimensions."
+      dimL = n1
+      @req length(s) == dimL "Invalid number of basis element names."
+      if check
```

Comment:
Apart from these checks, the implementation of `AbstractLieAlgebra` (data structure, arithmetic operations) seems to allow for arbitrary finite-dimenaional algebras, not just Lie algebras.

Reply:
There is no multiplication of elements available here, only a lie `bracket`. Apart from that, that is true.